### DISCUSSION TITLE
Headless mode

### DISCUSSION THREAD

**Discussion by nandorojo** [add2cal/add-to-calendar-button#443] [Ideas / Requests]
Hey, any shot you could export the functions to generate the ics files (or whatever is required)? I'd love to use those, but I don't want to use any UI provided from the library itself.

Thanks!

**Comment by jekuer** [comment]
Hi @nandorojo ,

We are currently working on a premium solution to get this covered. Where you then can easily add stuff to emails, etc.

For the free library, it will always stay client focused by design. You can use the atcb_action option and define only the ical option. This will trigger the ics download for the user immediately. But it may not be used to generate stuff on the server side.

All the best,
Jens

**Reply by rylantalerico** [reply]
Hi, any update on headless mode?

**Reply by jekuer** [reply]
We just recently launched our PRO offering, which includes the option to generate ics files via API 🚀 😃 .
You can check it out via https://add-to-calendar-pro.com.

Regarding the open source tool:
This feature is currently not on the roadmap, as it comes with a major refactoring (if we want to do it right).
We will evaluate the success of the PRO version first, as this somehow indirectly pays for further bigger development steps.
After that, we are planning for this general refactoring, where the feature request will get some attention for sure. Still, there will be some bigger discussions under the hood as well. Like, how can we make sure it is only used directly and not to power a managed service (as it would be not allowed per license agreement). There sure is a way (and already ideas)!
All I want to point out here: It will take some time for this to happen, unfortunately.
